### PR TITLE
added explanation around BUFFER_QUEUE_LIMIT variable

### PR DESCRIPTION
**Pull request by gaurav-nelson**
For https://bugzilla.redhat.com/show_bug.cgi?id=1588430

### CODE REVIEW

File: admin_guide/overcommit.adoc [hunk 1]
```diff
@@ -74,7 +74,23 @@ filter-pre-mux-client.conf:
...
 ----
 
-The Fluentd `buffer_queue_limit` is 32.
+The Fluentd `buffer_queue_limit` is the value of the variable `BUFFER_QUEUE_LIMIT`. This value is `32` by default.
+
+The environment variable `BUFFER_QUEUE_LIMIT` is calculated as `(FILE_BUFFER_LIMIT / (number_of_outputs * BUFFER_SIZE_LIMIT))`.
+
+In this formula, `number_of_outputs` is `1` if all the logs are sent to a single resource and it is incremented by `1` for each additional resource. For example, the value of `number_of_outputs` will be:
+
+ * `1` - If all logs are sent to a single ElasticSearch node.
```
Comment (nhosoi) [inline admin_guide/overcommit.adoc]: This is a terminology issue, so I might be wrong.  But when I hear an "ElasticSearch node", I have an impression the ElasticSearch is running on a node, not in a container...  In the Common Logging EFK stack, each component is containerized.  In such case, how do you usually name it?

File: admin_guide/overcommit.adoc [hunk 2]
```diff
@@ -74,7 +74,23 @@ filter-pre-mux-client.conf:
...
+ * `2` - Incremented by `1`, if ops ElasticSearch is additionally configured.
+ * `3` - Incremented by `1` again, if the logs are also forwarded to other Fluentd.
```
Comment (nhosoi) [inline admin_guide/overcommit.adoc]: This number_of_outputs is a bit tricky. :)  Not just the 1, 2, 3, but there could be more combinations.  
Single ElasticSearch for all -- 1 output
One ElasticSearch for ops and another for application logs -- 2 outputs
One ElasticSearch for all and forward them to other Fluentd -- 2 outputs
One ElasticSearch for ops and another for application logs plus forward one of the outputs to other Fluentd -- 3 outputs
One ElasticSearch for ops and another for application logs plus forward the both outputs to other Fluentd -- 4 outputs
Please note that I'm not suggesting to list up all of them, but I'd rather think `2` `3` is not needed.  Or do you have any good idea to explain this?  Thanks!

File: admin_guide/overcommit.adoc (hunk 1 above)
Comment (gaurav-nelson) [reply admin_guide/overcommit.adoc]: Thanks @nhosoi for looking at this. I will update this to `pod`. Which is consistent with rest of our docs.

**Comment by gaurav-nelson** [comment]
@nhosoi I have updated the PR to address your comments. PTAL 🔢

**Comment by nhosoi** [comment]
@gaurav-nelson , thanks for the updates.
/lgtm

File: admin_guide/overcommit.adoc [hunk 3]
```diff
@@ -74,7 +74,25 @@ filter-pre-mux-client.conf:
   buffer_path `/var/lib/fluentd/buffer-mux-client`
 ----
 
-The Fluentd `buffer_queue_limit` is 32.
+The Fluentd `buffer_queue_limit` is the value of the variable `BUFFER_QUEUE_LIMIT`. This value is `32` by default.
+
+The environment variable `BUFFER_QUEUE_LIMIT` is calculated as `(FILE_BUFFER_LIMIT / (number_of_outputs * BUFFER_SIZE_LIMIT))`.
+
+In this formula, `number_of_outputs` is `1` if all the logs are sent to a single resource and it is incremented by `1` for each additional resource. For example, the value of `number_of_outputs` will be:
```
Comment (kalexand-rh) [inline admin_guide/overcommit.adoc]: s/resource/resource,
s/will be/is

File: admin_guide/overcommit.adoc [hunk 4]
```diff
@@ -74,7 +74,25 @@ filter-pre-mux-client.conf:
...
+
+ * `1` - If all logs are sent to a single ElasticSearch pod.
```
Comment (kalexand-rh) [inline admin_guide/overcommit.adoc]: I'd reformat the list entries: s/- If/if
I'd also remove the ending periods.

File: admin_guide/overcommit.adoc [hunk 5]
```diff
@@ -74,7 +74,25 @@ filter-pre-mux-client.conf:
...
+ * `2` - If application logs are sent to an ElasticSearch pod and ops logs are sent to
+another ElasticSearch pod.
+ * `4` - If application logs are sent to an ElasticSearch pod and ops logs are sent to
```
Comment (kalexand-rh) [inline admin_guide/overcommit.adoc]: Maybe "If application logs are sent to an ElasticSearch pod, ops logs are sent to
another ElasticSearch pod, and all logs are then forwarded to other Fluentd instances"

File: admin_guide/overcommit.adoc [hunk 6]
```diff
@@ -74,7 +74,25 @@ filter-pre-mux-client.conf:
...
+another ElasticSearch pod. And both are forwarded to other Fluentd instances.
+
+Therefore, if `BUFFER_QUEUE_LIMIT` has the default set of values:
```
Comment (kalexand-rh) [inline admin_guide/overcommit.adoc]: Are these the default values? I'd say "By default, these parameters are assigned the following values:"

I'd move lines 89-95 to line 80, right after you present the formula.

File: admin_guide/overcommit.adoc [hunk 7]
```diff
@@ -74,7 +74,25 @@ filter-pre-mux-client.conf:
...
+
+* `FILE_BUFFER_LIMIT = 256Mi`
+* `number_of_outputs = 1`
+* `BUFFER_SIZE_LIMIT = 8Mi`
+
+The value of `buffer_queue_limit` will be `32`. To change the `buffer_queue_limit`, you need to change `FILE_BUFFER_LIMIT`.
```
Comment (kalexand-rh) [inline admin_guide/overcommit.adoc]: s/The value of `buffer_queue_limit` will be `32`.//
s/`FILE_BUFFER_LIMIT`./`FILE_BUFFER_LIMIT` value or the number of outputs.

**Review by kalexand-rh — COMMENTED** [review]
Couple of suggestions to consider and a few nits. This sounds good!

**Comment by gaurav-nelson** [comment]
@kalexand-rh I have made suggested changes. Except one, haven't changed the following:
The value of `buffer_queue_limit` will be `32`. To change the `buffer_queue_limit`, you need to change the value of `FILE_BUFFER_LIMIT`.

**Comment by kalexand-rh** [comment]
Sounds good to me! Thanks @gaurav-nelson!

**Pull request merged**